editor/gui/editor_run_bar.h

```diff
@@ -109,6 +111,8 @@ class EditorRunBar : public MarginContainer {
 
 	Button *get_pause_button() { return pause_button; }
 
+	Control *get_buttons_container();
```

Comment:
Yeah, exactly. Though even if it was, not reporting the correct type make it harder to use reliably. It would allow us to change the underlying type without formally breaking compat, but then on the user end it could still break compat if they expect it to be HBox and we changed it for something else.

Reply:
If the API is returning `Control` you can't assume it'll be `HBox`. To me that's like using paths to internal editor nodes, it may work but it's not guaranteed. That the `Control` happens to be `HBox` is just an implementation detail, the user is able to use whatever API is available on `Control` and casting to a more concrete type is like accessing the internals.

Reply:
Well, the method returns `HBoxContainer`, we would just be telling that it's something else. It's not some hidden information, we expose this class completely with all of its methods and properties when we return the element. In fact, it being a container, a specific container can be very important if you need to change size flags to make your element display correctly.

Basically, if we expose it to the API, it's no longer internal. So IMO we can't apply the same logic.

Reply:
A `HBoxContainer` is still a `Control`, we would just be returning a less concrete type. For users to get the `HBoxContainer` they'd have to cast it, but nothing guarantees that the cast will work. If you try to cast the `Control` to `Button` it will fail.

> In fact, it being a container, a specific container can be very important if you need to change size flags to make your element display correctly.

That's true, but if the return type is `Control` you can't assume it's a container either, so you can't reliably change the size flags since you can't know how it will behave. I think I'm starting to see what you mean, yeah it's probably a good idea to return the most concrete type when it comes to containers.